crates/pipeline_manager/src/api/service.rs

```diff
@@ -84,6 +87,27 @@ pub(crate) struct CreateOrReplaceServiceResponse {
     service_id: ServiceId,
 }
 
+/// Response to a create service probe request.
+#[derive(Serialize, ToSchema)]
+pub(crate) struct CreateServiceProbeResponse {
+    /// Unique id assigned to the service probe.
+    service_probe_id: ServiceProbeId,
+}
+
+/// Request to retrieve a (limited) list of service probes, optionally filtered
+/// by id.
+#[derive(Debug, Deserialize, IntoParams)]
+pub struct ListServiceProbes {
+    /// If provided, will filter based on exact match of the service probe
+    /// identifier.
+    id: Option<Uuid>,
+    /// If provided, will limit the amount of probes to the N most recent.
+    limit: Option<u32>,
+    /// If provided, will only have probes of that particular type.
+    #[serde(rename = "type")]
+    request_type: Option<String>,
```

Comment:
This should be an enum like `type: Option<ProbeType>`. And using a sharper type will allow us to generate better documentation too (swagger can list the types of probes and what they do for example).

Reply:
Yes, would be nice to have the full enumeration list in the generated API doc, will investigate and add

Reply:
I added a `ServiceProbeType` enumeration (required quite some additional code to have it be consistent both at API and database level).